### PR TITLE
Use base_footprint for landing detection instead of quad

### PR DESCRIPTION
**Pull request by amiller27**

### CODE REVIEW

**Comment by amiller27** [comment]
Tested in sim, in theory this should work just fine on all other platforms.  Configs for other platforms were shifted by offset from `quad` to `base_footprint` just like the one on the sim drone

**Review by aftersomemath — APPROVED** [review]

**Comment by aftersomemath** [comment]
This has broken startup. It depends on transforms which aren't published unless nodes higher on the safety response start sending sensor data to robot localization.

File: src/iarc7_sensors/landing_detector.py [hunk 1]
```diff
@@ -74,6 +79,14 @@ def odom_callback(msg):
                 rospy.logwarn_throttle(1.0,
                     'Landing Detector is not receiving switch messages fast enough')
 
+        # If this fails something is seriously wrong, so we're ok with crashing
```
Comment (aftersomemath) [inline src/iarc7_sensors/landing_detector.py:82]: This is not ok because then fc_comms will not be able to execute a safety landing.

Comment (amiller27) [reply src/iarc7_sensors/landing_detector.py:82]: No this shouldn't fail unless there's memory corruption in the node or something, or we haven't received a transform for so long that the internal buffer in tf is empty, and I'm not even sure if that's possible

**Comment by amiller27** [comment]
The transform from `base_footprint` to `quad` is static, it should be available immediately

**Comment by aftersomemath** [comment]
I know this is causing a bug because reverting this commit just took V1.5 from not be able to start nodes to starting nodes. It was halting while waiting for map->base_footprint which depends on level_quad

**Comment by amiller27** [comment]
Ah that means this uncovered a bug in our start order and we were previously starting up unsafely!

**Comment by aftersomemath** [comment]
It was previously handled by assuming the Quad was on the ground at startup. Low level motion also follows this assumption.

**Comment by aftersomemath** [comment]
Also, this node doesn't fit into the safety scheme unless the altimeter nodes (Which are way more important) are also brought in.

**Comment by amiller27** [comment]
The correct way to fix this would be for fc_comms to start publishing before forming its bond, but not to accept commands. Not sure if that's too much work

**Comment by aftersomemath** [comment]
A lot of work due to the way the safety client and fc comms are designed, thought I agree a better policy is for nodes to publish sensor data regardless of the safety monitor status. I think this node should just assume we are on the ground with a timeout for receiving a transform. This fits the assumptions used throughout the rest of the stack.

**Comment by amiller27** [comment]
Actually I just realized this was previously dependant on `level_quad`, not `quad`, so there was no circular dependency before. We can revert this.

**Comment by aftersomemath** [comment]
The circular dependency is causing trouble due to waiting on the transform tree during startup, not the transform itself.

**Comment by amiller27** [comment]
Yes I understand. We could revert this and add a check for odometry on startup and we'd be ok

**Comment by aftersomemath** [comment]
Assuming it waited for an odometry message the same problem would occur.

**Comment by aftersomemath** [comment]
An initial state has to be assumed or the architecture of something has to be switched up.

**Comment by amiller27** [comment]
Waiting for odometry should be fine, right?  robot_localization doesn't wait for anything, so it should come up without any dependency loops.

**Comment by aftersomemath** [comment]
Robot localization waits for at least one sensor message. It receives none if this thing halts. Altimeters cant publish poses without a transform tree. Neither can the camera node. And fc comms doesnt publish anything until its safety bond is formed.

I like using the transform tree in this node. But it just needs to assume we are on the ground when it starts up.

**Pull request merged**